### PR TITLE
fix: use returnDocument after option in update

### PR DESCRIPTION
**Pull request by ghost**
`returnOriginal: false` is deprecated after mongodb 3.6, so when using the library with newer clients, the option is not working.
Instead, the `returnDocument: 'after'` option is compatible with all versions.

http://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html#findOneAndUpdate
https://stackoverflow.com/questions/35626040/how-to-get-updated-document-back-from-the-findoneandupdate-method

### CODE REVIEW

**Comment by pierissimo** [comment]
hey @adriasala124, thank you for the change.
Is that in relation to the DaVinci migration in the Dashboard API?

**Comment by ghost** [comment]
> hey @adriasala124, thank you for the change. Is that in relation to the DaVinci migration in the Dashboard API?

hey @pierissimo! yes! This is what was causing trouble last time. The documents were expected to be returned after the update but they weren't.

**Comment by pierissimo** [comment]
@adriasala124 ok good spot. 
Are the test passing? I don't have visibility on the forked repo on the oneflow org

**Comment by ghost** [comment]
> @adriasala124 ok good spot. Are the test passing? I don't have visibility on the forked repo on the oneflow org

Thanks :) There was a test failing but it was before my changes already, and seems completely unrelated.

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,7 +8,6 @@ workflows:
     jobs:
       - queue/block_workflow:
           context: pierissimo
-          only-on-branch: master
```
Comment (pierissimo) [inline .circleci/config.yml]: we can probably revert this

File: src/lib/Queue.ts [hunk 2]
```diff
@@ -242,7 +242,7 @@ export default class Queue extends EventEmitter {
     };
 
     const msg = await this.collection.findOneAndUpdate(query, update, {
-      returnOriginal: false
+		returnDocument: 'after'
```
Comment (pierissimo) [inline src/lib/Queue.ts]: those lines have a weird indentation (and use spaces rather than tabs)

**Review by pierissimo — APPROVED** [review]

**Comment by ghost** [comment]
Thank you very much @pierissimo!

**Comment by pierissimo** [comment]
Thank you!

**Comment by ghost** [comment]
Sorry @pierissimo! The release is not being triggered... Is there anything missing?

**Comment by pierissimo** [comment]
Taking a look

**Comment by pierissimo** [comment]
hey @adriasala124 I feel the returnDocument option doesn't always work as expected.
Different tests are failing due to findOneAndUpdate returning the old document.
I've opened a new branch 'chore/circleci-and-tests' to clean up the circleci pipeline but also to refactor some tests.

**Comment by pierissimo** [comment]
I've added a failing test here: https://github.com/pierissimo/monkeycymbal/blob/chore/circleci-and-tests/test/queue/default.ts#L22

**Comment by ghost** [comment]
I see. I think there is a breaking change there. Should we update to mongodb 4 or later @pierissimo ?

**Comment by pierissimo** [comment]
Maybe it's a good idea, but a lot of places in the codebase would need to be adjusted

**Comment by pierissimo** [comment]
@adriasala124 started working on this branch: https://github.com/pierissimo/monkeycymbal/tree/refactor/mongodb-4

**Comment by pierissimo** [comment]
@adriasala124 please have a look at this: https://github.com/pierissimo/monkeycymbal/pull/23

**Comment by pierissimo** [comment]
version 1.2.0 has been published.

**Comment by juanjoseruiz** [comment]
Thanks @pierissimo for the support. Everything seems to be working fine with the new version :)

**Comment by pierissimo** [comment]
Glad to hear that!

**Pull request merged**